source/firstprogram.rst

```diff
@@ -1,109 +1,146 @@
 
+.. note:: It can be named as you please. For the sake of clarity, it has been named as **Person** here
 
-It is important that one defines the Behaviours in the order that they are to be compiled. This is based on the epidemiological SIR model, where an agent transitions from one compartment to another in a chronological manner, from S to I to R, as shown below.
+2. Similar to the ``House`` case class described above, the ``Person`` case class is defined by a set of attributes. These attributes are generally the characteristics of a generic person like a person id, age etc. To define the Person case class, one must also call its attributes, which in this case are the id and age. 
 
-.. image:: _static/images/SIR-compartment.png
+.. code-block:: scala
+    
+  case class Person(id: Long, age: Int) extends Agent {
+  }
+    
+3. In order to add the relationship between the Person and the components of the Network, write the following code within the case class Person.
 
-After the behaviours are defined, they need to be added to the simulation in the order in which they are executed. This is done using the framework defined ``addBehaviour`` function.
+.. code-block:: scala
+  
+  addRelation[House]("STAYS_AT")
+  addRelation[Office]("WORKS_AT")
+  addRelation[School]("STUDIES_AT")
+
+4. Given below is an example which will help you to understand the importance of attributes as well as behaviours. Consider the year ‘1984’. During this time, Big Brother doesn’t allow people below the age of 25 to watch ‘Harry Potter’ movies. To model this scenario, you can add a parameter ‘canIWatchHarryPotter’ when defining the ``Person`` case class and let it’s default value be “No”.
 
 .. code-block:: scala
+  
+  import com.bharatsim.engine.Context
+      
+  case class Person(id:Long, age:Int, canIWatchHarryPotter = ‘No’: String) extends Agent
 
-   addBehaviour(incrementInfectionDay)
-   addBehaviour(checkForInfection)
-   addBehaviour(checkForRecovery)
+.. note:: String is a data-type which takes strings as the arguments. 
 
-Each agent goes through the above-mentioned behaviours chronologically during each tick. For example, if there are 100 agents in the simulation, all 100 of them go through the behaviours as listed chronologically above, and this process repeats at each tick.
 
+Assume that the name of this behaviour is ``watchMovie``. So, the task of the behaviour is to change the value of the parameter ``canIWatchHarryPotter`` from ‘No’ to ‘Yes’ for people above the age of 25. 
 
+.. note:: The behaviour takes ``Context`` as an argument so it has to be imported.
+
+
+This can be done using the framework defined ``updateParam`` function which updates the specified parameters. The function takes two arguments, the parameter which is to be updated and the updated value.  
+
+.. code-block:: scala
+  
+  val watchMovie : Context => Unit = (context:Context) => {
+      if (age >= 25) {
+          updateParam("canIWatchHarryPotter", ‘Yes’)}

```

Comment:
Make it a `this.updateParam`. 

Besides, this is not a very good behaviour, since it only happens once since the age doesn't change. It's not necessary to call this beyond the first tick!